### PR TITLE
v2

### PR DESCRIPTION
**Pull request by josephbuchma**
Compiles with https://github.com/db-journey/migrate/pull/3

# UPDATE
Now all original behavior is restored (including progress logging, ctrl-c to stop & repeat to force stop)
Also got rid of migrate "shortcut" functions.

## New features
- apply specific version (run up migration)
- rollback specific version (run down migration)

### CODE REVIEW

**Comment by gravis** [comment]
LGTM.
I'm reviewing all projects before merging anything. Btw, db-journey already hit v2 because of incompatible changes, so it's going to be v3 :)

**Comment by gravis** [comment]
hmm, the api doesn't change here, no need for v3, ignore my last comment.

**Comment by josephbuchma** [comment]
It still needs small updates, like bringing back migrations progress logging, or "press Ctrl-c to interrupt immediately" etc.
I'll wait until you review everything to be sure you're ok with what I've done so far.

**Comment by gravis** [comment]
Thanks for all your changes.
I have a lot of things ongoing this week, and will do my best to merge everything before next week.
Thanks for your patience.

**Comment by josephbuchma** [comment]
Now all original behavior is restored (including progress logging, ctrl-c to stop & repeat to force stop)
Also got rid of migrate "shortcut" functions.

### New features

apply specific version (run up migration)
rollback specific version (run down migration)

**Comment by gravis** [comment]
You're changing too many things at once, it's becoming harder and harder to merge... :)

File: commands/migrate_commands.go
Comment (gravis) [inline commands/migrate_commands.go:298]: A lib should never presume of context behaviour.
You must pass a `context` to migrate, and handle this stuff in https://github.com/db-journey/journey/ instead.
That way, you let the users chose where `context.Background()` is created.

File: commands/migrate_commands.go
Comment (gravis) [inline commands/migrate_commands.go:274]: So, why the `context.Context` is not a field of `*migrate.Handle`?
I smell a wrong utilization of context here.

**Review by gravis — CHANGES_REQUESTED** [review]
context is the responsibility of the caller.
Apart from that, we're good to go

File: commands/migrate_commands.go
Comment (josephbuchma) [reply commands/migrate_commands.go:274]: It shouldn't be a field of `*migrate.Handle`. If you add context to Handle at create time, it will make this Handle useless once you cancel this context.

File: commands/migrate_commands.go
Comment (josephbuchma) [reply commands/migrate_commands.go:298]: I thought that `commands` package is not supposed to be used apart from journey CLI. And it doesn't presume context behavior. The only thing it "presumes" is that context _might_ be canceled, which is kind of normal thing to presume when you work with context.
So passing context here from `journey` or whatever else won't change/break anything.

Comment (josephbuchma) [reply commands/migrate_commands.go:298]: If you talking about os signals processing - yes, it does makes sense to move it to journey's main package if `commands` is used elsewhere beyond journey.

Comment (gravis) [reply commands/migrate_commands.go:298]: my bad, I mixed up with the `migrate` project.
You're right!

**Pull request merged**
